pkg/cloudprovider/providers/vra/vra_loadbalancer.go

```diff
@@ -0,0 +1,291 @@
+
+
+package vra
+
+import (
+	"io/ioutil"
+	"bytes"
+	"context"
+	"fmt"
+	"reflect"
+	"strings"
+	"encoding/json"
+	"net/http"
+	"k8s.io/klog"
+	"k8s.io/api/core/v1"
+	cloudprovider "k8s.io/cloud-provider"
+	"strconv"
+)
+
+const (
+
+	ServiceAnnotationProjectID    = "loadbalancer.vra.org/project-id"
+
+	// ServiceAnnotationLoadBalancerInternal is the annotation used on the service
+	// to indicate that we want an internal loadbalancer service.
+	// If the value of ServiceAnnotationLoadBalancerInternal is false, it indicates that we want an external loadbalancer service. Default to false.
+	ServiceAnnotationLoadBalancerInternetFacing = "service.beta.kubernetes.io/vra-internet-facing-load-balancer"
+	ServiceAnnotationLoadBalancerNicDeviceIndex = "service.beta.kubernetes.io/vra-load-balancer-nics-index"
+	ServiceAnnotationLoadBalancerNetworkId = "service.beta.kubernetes.io/vra-load-balancer-network-id"
+	ServiceAnnotationLoadBalancerAddress = "service.beta.kubernetes.io/vra-load-balancer-address"
+	// ServiceAnnotationLoadBalancerSecurityGroups = "service.beta.kubernetes.io/vra-load-balancer-security-group"
+	ServiceAnnotationLoadBalancerTargetLinks = "service.beta.kubernetes.io/vra-load-balancer-target-links"
+
+	ServiceAnnotationLoadBalancerHealthyThreshold = "service.beta.kubernetes.io/vra-load-balancer-healthy-threshold"
+	ServiceAnnotationLoadBalancerUnHealthyThreshold = "service.beta.kubernetes.io/vra-load-balancer-unhealthy-threshold"
+	ServiceAnnotationLoadBalancerTimeOut = "service.beta.kubernetes.io/vra-load-balancer-timeout"
+	ServiceAnnotationLoadBalancerIntervalSeconds = "service.beta.kubernetes.io/vra-load-balancer-interval-seconds"
+	ServiceAnnotationLoadBalancerUrlPath = "service.beta.kubernetes.io/vra-load-balancer-urlpath"
+
+)
+
+
+type  VraTokenResp struct {
+	TokenVal string `json:"token"`
+}
+
+
+type LbaasV2 struct {
+	LoadBalancer
+}
+
+type empty struct{}
+
+
+func getSecurityGroupName(service *v1.Service) string {
+	securityGroupName := fmt.Sprintf("lb-sg-%s-%s-%s", service.UID, service.Namespace, service.Name)
+	if len(securityGroupName) > 255 {
+		securityGroupName = securityGroupName[:255]
+	}
+	return securityGroupName
+}
+
+// GetLoadBalancer returns whether the specified load balancer exists and its status
+func (lbaas *LbaasV2) GetLoadBalancer(ctx context.Context, clusterName string, service *v1.Service) (*v1.LoadBalancerStatus, bool, error) {
```

Comment:
Do we need a LbaasV2 here as well? This was added for versioning the different implementations I believe